mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRV.cpp

```diff
@@ -597,6 +606,120 @@ class GPUSubgroupReduceConversion final
   }
 };
 
+/// Pattern to convert a gpu.printf op into a SPIR-V CLPrintf op.
+
+LogicalResult GPUPrintfConversion::matchAndRewrite(
+    gpu::PrintfOp gpuPrintfOp, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  Location loc = gpuPrintfOp.getLoc();
+
+  auto moduleOp = gpuPrintfOp->getParentOfType<spirv::ModuleOp>();
+
+  if (!moduleOp) {
+    return success();
+  }
```

Comment:
`success` indicates that you matched, so we need to return `failure` here. Also some stylistic nits:
```suggestion
  auto moduleOp = gpuPrintfOp.getParentOfType<spirv::ModuleOp>();
  if (!moduleOp)
    return failure();
```